<?xml version="1.0"?>
<launch>
<arg name="DEVICE" default="/dev/video9"/>
  <!-- The GStreamer framerate needs to be an integral fraction -->
  <arg name="FPS" default="30/1"/>

  <node pkg="nodelet" type="nodelet"
    name="standalone_nodelet" args="manager"
    output="screen"/>

  <node pkg="nodelet" type="nodelet"
    name="GSCamNodelet"
    args="load gscam/GSCamNodelet standalone_nodelet"
    output="screen">
    <param name="camera_name" value="default"/>
    <param name="camera_info_url" value="file://$(find gscam)/calibration/camera_calibrated"/>
    
    <param name="gscam_config" value="v4l2src device=$(arg DEVICE) ! video/x-raw-rgb,framerate=$(arg FPS) ! ffmpegcolorspace"/>
    <param name="frame_id" value="/v4l_frame"/>
    <param name="sync_sink" value="true"/>
    
    </node>
    <!--
<env name="GSCAM_CONFIG" value="v4l2src device=/dev/video0 ! video/x-raw-rgb ! ffmpegcolorspace"/>
	<node ns="bingo_camera" pkg="gscam" type="gscam" 		name="gscam" output="screen" cwd = "node">
	
	
		<param name="width" type="int" value="640" />
		<param name="height" type="int" value="480" />
		<param name="FPS" type="int" value="30" />
		<param name="frame" type="string" value="webcam" />
		<param name="device" type="string" value="/dev/video0" />
		-->
	

	<node pkg="bingo_detection" type="bingo_detection_server" name="bingo_detection_server" output="screen">
		<param name="cardDatabaseFolder" type="string" value="$(find bingo_detection)/Marker_Database/" />
		<param name="numberDatabaseFile" type="string" value="$(find bingo_detection)/Numbers.txt" />
		<remap from="in" to="/camera/image_raw"/>
    <remap from="out" to="/bingo_detection/analyzed_image" />
	</node>
	
	 <!--  -->
</launch>






